docs/src/main/asciidoc/spring-cloud-sleuth.adoc

```diff
@@ -273,10 +274,117 @@ the number of dropped spans will get increased.
+----
+
+=== HTTP integration
+
+Features from this section can be disabled by providing the `spring.sleuth.web.enabled` property with value equal to `false`.
+
+==== HTTP Filter
+
+Via the `TraceFilter` all sampled incoming requests result in creation of a Span. That Span's name is `http:` + the path to which
+ the request was sent. E.g. if the request was sent to `/foo/bar` then the name will be `http:/foo/bar`.
+
+You can configure which URIs you would like to skip via the `spring.sleuth.instrument.web.skipPattern` property.
+
+==== Async Servlet support
+
+If your controller returns a `Callable` or a `WebAsyncTask` Spring Cloud Sleuth will continue the existing span instead of creating a new one.
+
+=== HTTP client integration
+
+In order to block the synchronous `RestTemplate` features just set `spring.sleuth.client.enabled` to `false`.
+To block the `AsyncRestTemplate` features set `spring.sleuth.async.client.enabled` to `false`.
+
+==== Synchronous Rest Template
+
+We're injecting a `RestTemplate` interceptor that ensures that all the tracing information is passed to the requests. Each time a
+call is made a new Span is created. It gets closed upon receiving the response.
+
+==== Asynchronous Rest Template
+
+Custom instrumentation is set to create and close Spans upon sending and receiving requests.
+
 === Feign
 
 By default Spring Cloud Sleuth provides integration with feign via the `TraceFeignClientAutoConfiguration`. You can disable it
 by setting `spring.sleuth.feign.enabled` to false.
 
 We're taking care of Feign instrumentation by means of a `FeignBeanPostProcessor`.
-You can disable the post processor by providing the `spring.sleuth.feign.processor.enabled` equal to `false`.
\ No newline at end of file
+You can disable the post processor by providing the `spring.sleuth.feign.processor.enabled` equal to `false`.
+
+=== Asynchronous communication
+
+In Spring Cloud Sleuth we're instrumenting async related components so that the tracing information is passed between threads. You can disable this behaviour
+by setting the value of `spring.sleuth.async.enabled` to `false`.
+
+==== @Async / @Scheduled annotated methods
+
+If you annotate your method with `@Async` / `@Scheduled` then we'll automatically create a new Span with the following characteristics:
+
+    - the Span name will be the annotated method name
+    - the Span will be tagged with that method's class name and the method name too
+
+==== Executor, ExecutorService and ScheduledExecutorService
+
+We're providing `LazyTraceExecutor`, `TraceableExecutorService` and `TraceableScheduledExecutorService`. Those implementations
+are creating Spans each time a new task is submitted, invoked or scheduled.
+
+=== Messaging
+
+To disable these features just set the `spring.sleuth.integration.enabled` property to `false`.
```

Comment:
nit: I think you have at least 3 sentence variants of this instruction. For each section, 

// <20 word overview and how to disable
Foo instrumentation (adds X, tags Y, etc). To disable Foo instrumentation, set `spring.sleuth.foo.enabled` to false.

// if more info is available, detail it in a new paragraph

Ex 1. Spring Integration
> http://projects.spring.io/spring-integration/[Spring Integration] instrumentation creates spans for publish and subscribe  events. To disable Spring Integration instrumentation, set `spring.sleuth.integration.enabled` to false.

Ex 2. Zuul
> https://github.com/Netflix/zuul/[Zuul] instrumentation propagates trace information on outbound via http headers. To disable Zuul instrumentation, set `spring.sleuth.zuul.enabled` to false.